app/main.py

```diff
@@ -1,3 +1,51 @@
+from __future__ import annotations
+
+
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: int | float) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: Distance | int | float) -> Distance:
+
+        return (Distance(self.km + other.km)
+                if isinstance(other, Distance)
+                else Distance(self.km + other))
+
+    def __iadd__(self, other: Distance | int | float) -> Distance:
+        self.km = ((self.km + other.km)
+                   if isinstance(other, Distance)
+                   else (self.km + other))
+        return self
+
+    def __mul__(self, other: int | float) -> Distance:
+        return Distance(self.km * other)
+
+    def __truediv__(self, other: int | float) -> Distance:
+        return Distance(round(self.km / other, 2))
+
+    def __lt__(self, other: Distance | int | float) -> bool:
+        if isinstance(other, Distance):
+            return self.km < other.km
+        return self.km < other
+
+    def __gt__(self, other: Distance | int | float) -> bool:
+        if isinstance(other, Distance):
+            return self.km > other.km
+        return self.km > other
+
+    def __eq__(self, other: Distance | int | float) -> bool:
+        if isinstance(other, Distance):
+            return self.km == other.km
+        return self.km == other
+
+    def __le__(self, other: Distance | int | float) -> bool:
+        return self.__eq__(other) or self.__lt__(other)
+
+    def __ge__(self, other: Distance | int | float) -> bool:
+        return self.__eq__(other) or self.__gt__(other)
```

Comment:
It is possible to make a comparison using `not` and `<` operator
```suggestion
        return not self < other
```